docs/gke-autopilot.md

```diff
@@ -0,0 +1,268 @@
+WARNING: Starting with version 1.19, newly created clusters and node-pools will have COS_CONTAINERD as the default node image when no image type is specified.
+Creating cluster auto-microservices-demo in us-central1...done.
+Created [https://container.googleapis.com/v1beta1/projects/my-project/zones/us-central1/clusters/microservices-demo].
+To inspect the contents of your cluster, go to: https://console.cloud.google.com/kubernetes/workload_/gcloud/us-central1/microservices-demo?project=my-project
+kubeconfig entry generated for microservices-demo.
+NAME                       LOCATION     MASTER_VERSION  MASTER_IP     MACHINE_TYPE  NODE_VERSION    NUM_NODES  STATUS
+microservices-demo         us-central1  1.20.8-gke.900  35.223.7.213  e2-medium     1.20.8-gke.900  3          RUNNING
+```
+You can safely ignore the warnings. These simply indicate the cluster was
+created with default values. Other than network settings, these values can't be
+overridden anyway for an Autopilot cluster.
+
+You can see that the cluster is initially configured with three
+conservatively-sized machines for the nodes. These nodes are distributed across
+three zones in the `us-central1` region.
+
+## Deploying the demo
+
+Now that you have an Autopilot cluster, you can deploy the Online Boutique using
+Kubernetes:
+
+```text
+kubectl apply -f release/kubernetes-manifests.yaml
+```
+
+The output should look similar to deploying the cluster in standard mode, with a
+notable exception: you may see warnings similar to this:
+
+```text
+Warning: Autopilot increased resource requests for Deployment default/checkoutservice to meet requirements. See http://g.co/gke/autopilot-resources.
+deployment.apps/checkoutservice created
+service/checkoutservice created
+```
+
+Again, this is nothing to be concerned about, but it is notable that Autopilot
+will automatically adjust pod resource requirements to fit within [allowable
+resource ranges].
+
+Once the application has finished deploying, you can get the public IP with the
+following command:
+
+```text
+PUBLIC_IP=$(kubectl get service frontend-external -o jsonpath="{.status.loadBalancer.ingress[0].ip}")
+```
+
+## Wrapup: Why use GKE Autopilot mode?
+
+Most developers are familiar with **standard mode** since **Autopilot mode** is
+a new mode that launched in early 2021.
+
+GKE Autopilot mode provides a *nodeless* experience that optimizes cluster node
+resource provisioning so that you can focus on your workloads instead of
+infrastructure and only pay for the pods you actually use. As application load
+increases, your Autopilot cluster will automatically scale as necessary.
+However, if any node resources provisioned to ensure availability are being
+underutilized, there's no need for concern -- you only pay consumption costs for
```

Comment:
**Optional:** Let's use — ([em dash](https://www.grammarly.com/blog/dash/)) instead of --.